src/vs/editor/contrib/rename/rename.ts

```diff
@@ -190,6 +193,8 @@ class RenameController implements IEditorContribution {
 			return undefined;
 		}
 
+		this._cts = new EditorStateCancellationTokenSource(this.editor, CodeEditorStateFlag.Position | CodeEditorStateFlag.Value);
```

Comment:
How about allowing to define a range in which position changes are OK? You could pass the rename-range into the ctor and test if the position is still within the range when selection or position changes happen

Reply:
I updated it. Since the token is used before the range is calculated, I also added a setter method. Could synchronization issues could arise in the future?